app/components/Signin/index.jsx

```diff
@@ -26,10 +26,16 @@ var Login = React.createClass({
         }
     },
     componentWillMount(){
-        this.props.setTitle("RhinoBird");
-    },
-    componentDidMount() {
-        this.refs.email.focus();
+        var query = this.context.router.getCurrentQuery();
+        console.log(query);
+        if (query.ticket) {
+            $.post('/platform/api/genius_coming', {ticket: query.ticket, sign: query.sign}).then((data) => {
+                LoginAction.updateLogin(data);
+                this.context.router.transitionTo(this.context.router.getCurrentQuery().target || "/");
+            });
+        } else {
+            this.props.setTitle("RhinoBird");
```

Comment:
Even when the URL doesn't  contain any query, the query object is not NULL.  So the usage of 'query.ticket' is safe.    

If ticket parameter exists, it's means this request is coming from Genius Center's callback. So the page will be redirect to User page, so set title in this page is meanless.